### PR TITLE
Draft: modem-manager: detect presence of modem-power driver

### PR DESCRIPTION
**Pull request by DylanVanAssche**
Quectel EG25-G modem in the Pine64 PinePhone and PinePhone Pro
can be booted with the eg25-manager (user space) or modem-power (kernel space).
We only support the eg25-manager for upgrading these modems.

CC: @superm1 

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — DISMISSED** [review]
Looks great, thanks.

**Comment by DylanVanAssche** [comment]
I don't have a kernel available with the `modem-power` driver so testing is a bit hard.
If the driver is present in the kernel, it provides a sysfs entry in `/sys/class/modem-power` like:

- `/sys/class/modem-power/modem-power/is_busy`
- `/sys/class/modem-power/modem-power/powered`
- `/sys/class/modem-power/modem-power/killswitched`
- ...

**Comment by superm1** [comment]
> I don't have a kernel available with the `modem-power` driver so testing is a bit hard. If the driver is present in the kernel, it provides a sysfs entry in `/sys/class/modem-power` like:
> 
> * ` /sys/class/modem-power/modem-power/device/is_busy`
> * `/sys/class/modem-power/modem-power/device/powered`
> * `/sys/class/modem-power/modem-power/device/killswitched`
> * ...
> 
> Source: https://github.com/megous/linux/blob/modem-5.17/drivers/misc/modem-power.c

yeah I think that's sufficient.

Fixup the CI failures and then we can merge this.

File: plugins/modem-manager/fu-mm-device.c [hunk 1]
```diff
@@ -1171,6 +1174,24 @@ fu_mm_device_get_firmware_version_mbim(FuDevice *device, GError **error)
...
 }
 
+static FuFirmware *
+fu_mm_device_prepare_firmware(FuDevice *device,
+		              GBytes *fw,
+		              FwupdInstallFlags flags,
+		              GError **error)
+{
+	/* detect presence of unsupported modem-power driver */
+	if (g_file_test (MODEM_POWER_SYSFS_PATH, G_FILE_TEST_EXISTS) {
```
Comment (superm1) [inline plugins/modem-manager/fu-mm-device.c:1184]: missing parenthesis

**Comment by DylanVanAssche** [comment]
My clang-formatter was acting up, fixed :)

**Review by superm1 — APPROVED** [review]
OK, as long as CI passes now LGTM.

**Review by hughsie — CHANGES_REQUESTED** [review]
I think these need to be device inhibits. E.g added in plugin.c when the device is added. I can do an example when back home. Just before the update is deployed is too late and will scare the user.

**Comment by superm1** [comment]
> I think these need to be device inhibits. E.g added in plugin.c when the device is added. I can do an example when back home. Just before the update is deployed is too late and will scare the user.

The thing about making it an inhibit is that a kernel module can (potentially) be loaded after the daemon starts or before it quits.  So unless you do an `inotify` on `/sys/class` how will you catch that?

**Comment by hughsie** [comment]
> So unless you do an inotify on /sys/class how will you catch that?

I think that's exactly what we should do. Alternate proposal in https://github.com/fwupd/fwupd/pull/4353

**Pull request closed without merging**